packages/dara-components/js/common/select/select.tsx

```diff
@@ -95,6 +97,22 @@ function Select(props: SelectProps): JSX.Element {
         return items;
     }, [items]);
 
+    /**
+     * Converts a value to an Item, or finds a matching Item in formattedItems based on value.
+     * @param val - The value to be converted or matched.
+     */
+    const toItem = (val: any): Item | undefined | null => {
+        // We set initially the item as an Item type with the value as the label, so that select can show a value even if not present in the list
+        let foundItem = typeof val === 'string' || typeof val === 'number' ? { label: String(val), val } : val;
+        // Tries to get matching item based on value from formattedItems
+        const matchingItem = formattedItems.find((item) => isItem(item) && String(item.value) === String(val));
+        // If a matching Item is found, update foundItem
+        if (matchingItem) {
+            foundItem = matchingItem;
+        }
```

Comment:
If this always takes priority I suggest putting this at the top of the function and early retrning. Then we can just return `foundItem` if it didn't match.

Reply:
fair point, have updated this